main/src/main/scala/sbt/Defaults.scala

```diff
@@ -1387,6 +1390,91 @@ object Defaults extends BuildCommon {
     }
     analysisResult.analysis
   }
+  lazy val previousAnalysis = Def.inputKey[String]("gets previous analysis")
+  lazy val previousAnalysisParser = Def.setting {
+    Parsers.StringBasic
+  }
+  lazy val previousAnalysisCommand = Command.single("previousAnalysis") { (state, definition) =>
+    val responseId = state.currentCommand.flatMap(_.execId)
+    state.log.warn(s"definition content: $definition, and id: $responseId")
+    import sbt.internal.langserver.TextDocumentPositionParams
+    import sbt.internal.langserver.codec.JsonProtocol._
+    import sjsonnew.support.scalajson.unsafe.{ Parser => JsonParser, Converter }
+    val input =
+      Converter.fromJsonUnsafe[TextDocumentPositionParams](JsonParser.parseUnsafe(definition))
+    state.log.err(input.toString)
+    val updatedState = Project.updateCurrent(state).reload
+    val aggrAnal = {
+      val e = Project.extract(updatedState)
+      val skey = (Keys.compile in Compile).scopedKey
+      val tasks = e.structure.data.scopes
+        .map { scope =>
+          e.structure.data.get(scope, skey.key)
+        }
+        .collect {
+          case Some(t) => Aggregation.KeyValue(skey, t)
+        }
+        .toSeq
+      state.log.warn(s"tasks $tasks")
+      val complete = Aggregation.timedRun(updatedState, tasks, sbt.std.Transform.DummyTaskMap(Nil))
+      state.log.warn("here")
+      val r = complete.results.toEither.toOption.map { results =>
+        state.log.warn("and there")
+        results.map(_.value)
+      }
+      r
+    }.getOrElse(Seq.empty)
+    state.log.warn(s"aggregated analysis $aggrAnal")
+
+    val analysis = {
+      val compile = Keys.compile in Compile
+      Project.runTask(compile, updatedState, true).collect {
```

Comment:
Is this invoking compilation task just to get to Analysis?

Reply:
fixed, previousCompile only is used